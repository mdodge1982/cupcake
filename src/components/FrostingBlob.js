import React, {Component} from 'react';
import './FrostingBlob.css';

class FrostingBlob extends Component {
	render() {
		let {xPos,yPos,width,height,angle} = this.props;
		let className = 'FrostingBlob';
		let rotation = -1*(angle-1.57);
		if(yPos===0){
			className += ' splat';
			width = height;
			rotation = 0;
			
		}
		return (
			<li className={className} style={{
				left: xPos+'px',
				bottom: yPos+'px',
				width: width+'px',
				height: height+'px',
				transform: 'rotate('+rotation+'rad)',
			}}></li>
		);
	}
}

export default FrostingBlob;
