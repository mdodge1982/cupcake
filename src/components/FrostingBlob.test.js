import React from 'react';
import renderer from 'react-test-renderer';
import FrostingBlob from './FrostingBlob';

const blob = {
	xPos: 22,
	yPos: 34,
	width: 45,
	height: 82,
	angle: 1.6
};
test('Initial render', () => {
	const component = renderer.create(
		<FrostingBlob {...blob} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Splats when landing at the bottom', () => {
	blob.yPos = 0;
	const component = renderer.create(
		<FrostingBlob {...blob} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
