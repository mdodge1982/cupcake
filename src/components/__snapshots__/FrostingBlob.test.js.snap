// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial render 1`] = `
<li
  className="FrostingBlob"
  style={
    Object {
      "bottom": "34px",
      "height": "82px",
      "left": "22px",
      "transform": "rotate(-0.030000000000000027rad)",
      "width": "45px",
    }
  }
/>
`;

exports[`Splats when landing at the bottom 1`] = `
<li
  className="FrostingBlob splat"
  style={
    Object {
      "bottom": "0px",
      "height": "82px",
      "left": "22px",
      "transform": "rotate(0rad)",
      "width": "82px",
    }
  }
/>
`;
